### PR TITLE
Ban importing aliased domains.

### PR DESCRIPTION
**Pull request by n3wscott**
Eventing had some bad merges where we almost shipped `github.com/knative/eventing/` imports in `0.9.0`.

Adding a test for Knative to never have this be possible again.

### CODE REVIEW

File: test/imports/banned_deps_test.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+	if !strings.Contains(dir, "vendor") {
+		t.Skipf("knative.dev/pkg not vendored, skipping...")
+	}
+
+	parts := strings.Split(dir, "vendor")
+
+	dir = fmt.Sprintf("%s%s%s", parts[0], string(filepath.Separator), "vendor")
+
+	err := filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
```
Comment (evankanderson) [inline test/imports/banned_deps_test.go]: Why not use [Build.ImportDir](https://golang.org/pkg/go/build/#Context.ImportDir) ?

Comment (evankanderson) [reply test/imports/banned_deps_test.go]: (That way, you could have a tool that checks that a go package spec does not have certain includes.)

Comment (n3wscott) [reply test/imports/banned_deps_test.go]: I don't want to check anything except the vendor dir. They are have bad thing on their import path and that is fine as long as it is not in our vendor dir.

**Comment by adrcunha** [comment]
I think this should be converted into a tool and added as a common build test step to all repos.

/lgtm
/approve

**Pull request merged**